crates/core/src/subscription/subscription.rs

```diff
@@ -360,41 +332,48 @@ impl IncrementalJoin {
         &self,
         updates: impl IntoIterator<Item = &'a DatabaseTableUpdate>,
     ) -> Option<(JoinSide, JoinSide)> {
-        let mut lhs_ops = Vec::new();
-        let mut rhs_ops = Vec::new();
+        let mut lhs_inserts = Vec::new();
+        let mut lhs_deletes = Vec::new();
+        let mut rhs_inserts = Vec::new();
+        let mut rhs_deletes = Vec::new();
+
+        // Partitions deletes of `update` into `ds` and inserts into `is`.
+        let partition_into = |ds: &mut Vec<_>, is: &mut Vec<_>, updates: &DatabaseTableUpdate| {
+            for update in &updates.ops {
+                if update.op_type == 0 { &mut *ds } else { &mut *is }.push(update.row.clone());
+            }
+        };
 
+        // Partitions all updates into the `l/rhs_insert/delete_ops` above.
```

Comment:
```suggestion
        // Partitions all updates into the `(l|r)hs_(insert|delete)_ops` above.
```
I'm pretty sure this isn't a correct Perl-style regex, but I still find it more readable.